complement/pipeline.yml

```diff
@@ -73,6 +73,53 @@ steps:
         - exit_status: 128
           limit: 3
 
+  - command:
+      # Build the Synapse Workers docker image, which is located at:
+      # https://github.com/matrix-org/synapse/blob/master/docker/Docker-workers
+      # Note that this image is based on matrixdotorg/synapse:latest, which we're just
+      # pulling from docker hub
+      - wget https://github.com/matrix-org/synapse/archive/master.tar.gz
+      - tar -xzf master.tar.gz
+      - docker build -t matrixdotorg/synapse:workers -f synapse-master/docker/Dockerfile-workers synapse-master/
+      # Build the accompanying Synapse workers Complement docker container. This specifically
+      # sets up a Synapse worker setup for Complement, and is located at:
+      # https://github.com/matrix-org/complement/blob/master/dockerfiles/SynapseWorkers.Dockerfile.
+      - docker build -t complement-synapse -f dockerfiles/SynapseWorkers.Dockerfile dockerfiles/
+      # Run the tests!
+      - COMPLEMENT_BASE_IMAGE=complement-synapse go test -v -tags synapse_blacklist ./tests
```

Comment:
this script could go in the complement repo, though it's not necessarily long enough to be worthwhile.

Reply:
I'd like to keep it in here for now, so we have one place for the steps in the Complement pipeline and Synapse pipeline.

The `wget` and `tar` steps could probably be combined into one line as well. If it starts ballooning then moving it sounds more plausible.

Reply:
> I'd like to keep it in here for now, so we have one place for the steps in the Complement pipeline and Synapse pipeline.

well, they're in separate pipeline files, so I'm not sure that's much of a reason. Still, it's fine for now.